src/components/geo-map-navigation/geo-map-navigation.component.ts

```diff
@@ -0,0 +1,212 @@
+import { TerritoryLevel } from '../../datasets/geo.types';
+import { fromEvent, merge, Subject } from 'rxjs';
+import { filter, takeUntil } from 'rxjs/operators';
+import { formatY, humanizeMeasureName, humanizeTerritoryLevel } from '../../utils/formatters';
+
+const { CONTINENT, SUBCONTINENT, COUNTRY, CITY } = TerritoryLevel;
+
+@Component({
+  selector: 'app-geo-map-navigation',
+  templateUrl: './geo-map-navigation.component.html',
+  styleUrls: ['./geo-map-navigation.component.scss'],
+})
+export class GeoMapNavigationComponent implements GeoMapNavigationPreference, OnInit, OnDestroy {
+  @ViewChild(ScreenReaderComponent, { static: true }) screenReaderComponent: ScreenReaderComponent;
+
+  @Input() enabled: boolean;
+
+  @HostBinding('attr.tabindex') private readonly tabindex = 0;
+
+  private destroy$ = new Subject();
+
+  constructor(
+    @Inject('host') private host: GeoMapComponent,
+    public elementRef: ElementRef<HTMLElement>,
+  ) {
+  }
+
+  get INSTRUCTIONS() {
+    return t(GEO_MAP_NAVIGATION.INSTRUCTIONS);
+  }
+
+  get data() {
+    return this.host.data;
+  }
+
+  get unit() {
+    return this.host.unit;
+  }
+
+  set unit(unit) {
+    this.host.unit = unit;
+  }
+
+  get filteringTerritory() {
+    return this.host.filteringTerritory;
+  }
+
+  set filteringTerritory(filteringTerritory) {
+    this.host.filteringTerritory = filteringTerritory;
+  }
+
+  get meta() {
+    return this.host.meta;
+  }
+
+  get keywordElement() {
+    return this.host.keywordRef.nativeElement;
+  }
+
+  get activeDatumIndex() {
+    return this.host.activeDatumIndex;
+  }
+
+  set activeDatumIndex(activeDatumIndex) {
+    this.host.activeDatumIndex = activeDatumIndex;
+  }
+
+  get activeMeasureIndex() {
+    return this.host.activeMeasureIndex;
+  }
+
+  set activeMeasureIndex(activeMeasureIndex) {
+    this.host.activeMeasureIndex = activeMeasureIndex;
+  }
+
+  get measureNames() {
+    return this.host.measureNames;
+  }
+
+  private get i18nActiveDatumArgs() {
+    const activeDatum = this.data[this.activeDatumIndex];
+    if (!activeDatum) {
+      return null;
+    }
+    const measureName = this.measureNames[this.activeMeasureIndex];
+    return {
+      territory: activeDatum.territory.name,
+      value: formatY(activeDatum.values[measureName]),
+      measure: humanizeMeasureName(measureName),
+    };
+  }
+
+  focus() {
+    this.elementRef.nativeElement.focus();
+  }
+
+  @HostListener('keydown', ['$event'])
+  async handleKeyDown($event: KeyboardEvent): Promise<boolean> {
+    const { key, shiftKey, altKey, ctrlKey, metaKey } = $event;
+    if (altKey || ctrlKey || metaKey || key === 'Tab') {
+      return false;
+    }
+    $event.preventDefault();
+    $event.stopPropagation();
+    switch (key) {
+      case '-':
+      case '_':
+        this.unit = Math.max(this.unit - 1, this.filteringTerritory?.level ?? 0);
+        return this.readOutUnitAndFilteringTerritory();
+      case '+':
+      case '=':
```

Comment:
Why would `_` & `=` fall through here?

Reply:
I wanted to make them work without needing to press SHIFT. I added comments to clarify.